catalog/nato/stanag4559/nato-stanag4559-source/src/main/java/org/codice/alliance/nato/stanag4559/source/Stanag4559Source.java

```diff
@@ -0,0 +1,942 @@
+        if (library == null) {
+            LOGGER.error("{} : Unable to initialize the library interface.", getId());
+        } else {
+            LOGGER.debug("{} : Initialized Library Interface", getId());
+        }
+    }
+
+    /**
+     * Initializes all STANAG 4559 mandatory managers:
+     * CatalogMgr
+     * OrderMgr
+     * DataModelMgr
+     * ProductMgr
+     */
+    private void initMandatoryManagers() {
+        try {
+            LibraryManager libraryManager = library.get_manager(CATALOG_MGR, accessCriteria);
+            setCatalogMgr(CatalogMgrHelper.narrow(libraryManager));
+
+            libraryManager = library.get_manager(ORDER_MGR, accessCriteria);
+            setOrderMgr(OrderMgrHelper.narrow(libraryManager));
+
+            libraryManager = library.get_manager(PRODUCT_MGR, accessCriteria);
+            setProductMgr(ProductMgrHelper.narrow(libraryManager));
+
+            libraryManager = library.get_manager(DATA_MODEL_MGR, accessCriteria);
+            setDataModelMgr(DataModelMgrHelper.narrow(libraryManager));
+
+        } catch (ProcessingFault | SystemFault | InvalidInputParameter e) {
+            LOGGER.error("{} : Unable to retrieve mandatory managers.", id, e);
+        }
+
+        if (catalogMgr != null && orderMgr != null & productMgr != null && dataModelMgr != null) {
+            LOGGER.debug("{} : Initialized STANAG mandatory managers.", getId());
+        } else {
+            LOGGER.error("{} : Unable to initialize mandatory mangers.", getId());
+        }
+    }
+
+    /**
+     * Obtains all possible views that the Federated Source can provide. EX: NSIL_ALL_VIEW, NSIL_IMAGERY
+     * According to ANNEX D, TABLE D-6, the passed parameter in get_view_names is an empty list(not used).
+     *
+     * @return an array of views
+     */
+    private void initServerViews() {
+        View[] views = null;
+        try {
+            views = dataModelMgr.get_view_names(new NameValue[0]);
+        } catch (ProcessingFault | SystemFault | InvalidInputParameter e) {
+            LOGGER.error("{} : Unable to retrieve views.", id, e);
+        }
+        if (views == null) {
+            LOGGER.error("{} : Unable to retrieve views.", id);
+        }
+        this.views = views;
+    }
+
+    /**
+     * Obtains all possible attributes for all possible views that the Federated Source can provide, and
+     * populates a sortableAttributes map, as well as resultAttributes map that will be used for querying
+     * the server.
+     * According to ANNEX D, TABLE D-6, the passed parameter in get_view_names is an empty list(not used).
+     *
+     * @return a map of each view and the attributes provided by the source for that view
+     */
+    private void initSortableAndResultAttributes() {
+        HashMap<String, String[]> resultAttributesMap = new HashMap<>();
+        HashMap<String, List<String>> sortableAttributesMap = new HashMap<>();
+
+        if (views == null | views.length == 0) {
```

Comment:
Also, might as well put this check before you initialize your `HashMap`s so you don't create them when you don't need them.